RecoTracker/FinalTrackSelectors/plugins/TrackTfClassifier.cc

```diff
@@ -0,0 +1,123 @@
+#include "RecoTracker/FinalTrackSelectors/interface/TrackMVAClassifier.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "getBestVertex.h"
+
+#include "TrackingTools/Records/interface/TfGraphRecord.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+#include "RecoTracker/FinalTrackSelectors/interface/TfGraphDefWrapper.h"
+
+struct TfDnnCache {
+  TfDnnCache() : session_(nullptr) {}
+  tensorflow::Session* session_;
+};
+
+namespace {
+  struct TfDnn {
+    TfDnn(const edm::ParameterSet& cfg)
+        : tfDnnLabel_(cfg.getParameter<std::string>("tfDnnLabel"))
+
+    {}
+    TfDnnCache* cache_ = new TfDnnCache();
+
+    ~TfDnn() {
+      if (cache_->session_) {
+        tensorflow::closeSession(cache_->session_);
+        delete cache_;
+      }
+    }
+
+    static const char* name() { return "TrackTfClassifier"; }
+
+    static void fillDescriptions(edm::ParameterSetDescription& desc) {
+      desc.add<std::string>("tfDnnLabel", "trackSelectionTf");
+    }
+
+    void beginStream() {}
+
+    void initEvent(const edm::EventSetup& es) {
+      if (!cache_->session_) {
+        edm::ESHandle<TfGraphDefWrapper> tfDnnHandle;
+        es.get<TfGraphRecord>().get(tfDnnLabel_, tfDnnHandle);
+        tensorflow::GraphDef* graphDef_ = tfDnnHandle.product()->getGraphDef();
+        cache_->session_ = tensorflow::createSession(
+            graphDef_, 1);  //The integer controls how many threads are used for running inference
+      }
+      session_ = cache_->session_;
+    }
+
+    float operator()(reco::Track const& trk,
+                     reco::BeamSpot const& beamSpot,
+                     reco::VertexCollection const& vertices) const {
+      Point bestVertex = getBestVertex(trk, vertices);
```

Comment:
here a copy is being made of the vertex, is this intended? If not, `const auto& bestVertex = ...` should be fine.